andela/knights-bn-backend
# #170947679 Send notification to the manager when request is edited

#### What does this PR do?
- Send notification to the user and his/her manager when a request is modified
#### Description of Task to be completed?
- When a requester modifies something on the trip request he/she has made, notification will be
pop up and notification panel will be updated to both requester and his/ her manager, the notification will be clickable and redirect the user to the edited trip request.

#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn    backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-recieve-notification-when-edit-170947679` 
- run `npm install` to install dependencies
- check if you have PostgresQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations 
- run `npm run createTables` to create tables in the new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have and use this URL to login `http://localhost:4000/api/v1/auth/login` and down below there is a default user you can use
```
{
   "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
```
- open this URL in your browser `http://localhost:4000/client/editRequest.html`
allow notification permission first by clicking on allow button on the browser alert which looks like below image 
<img width="306" alt="Screen Shot 2020-03-17 at 07 56 30" src="https://user-images.githubusercontent.com/36619897/76826377-04b0dd80-6825-11ea-938c-5d2ea3662463.png">

 and see on the left side of the page, all your trip requests and choose one of them and update it by filling that form on that page.

- After filling the form click the send button and you will see the notification popping up and left side column will be updated, once you click on the notification you will be redirected to the modified trip request.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947679](https://www.pivotaltracker.com/story/show/170947679)
#### Screenshots
#### Notification
<img width="459" alt="Screen Shot 2020-03-13 at 09 34 38" src="https://user-images.githubusercontent.com/36619897/76599597-0d946d00-650e-11ea-8248-2fa895e38e25.png">

#### Update notification panel
<img width="1098" alt="Screen Shot 2020-03-13 at 09 14 45" src="https://user-images.githubusercontent.com/36619897/76599397-9f4faa80-650d-11ea-8fc1-f88c9736dd27.png">

#### Redirect to the trip request
<img width="1293" alt="Screen Shot 2020-03-13 at 09 26 12" src="https://user-images.githubusercontent.com/36619897/76599445-b4c4d480-650d-11ea-82a5-b7e7600fe12f.png">
